astropy/cosmology/parameter/_core.py

```diff
@@ -261,12 +296,12 @@ def register_validator(key, fvalidate=None):
 
     # -------------------------------------------
 
-    def clone(self, **kw):
+    def clone(self, **kw: Any) -> Self:
         """Clone this `Parameter`, changing any constructor argument.
 
         Parameters
         ----------
-        **kw
+        **kw : Any
```

Comment:
Apparently the [NumPy docstring standard](https://numpydoc.readthedocs.io/en/latest/format.html#parameters) says this should be written as
```suggestion
        **kw : dict, optional
```

Reply:
I think this is a mistake by the NumPy docstring standard. It certainly disagrees with how `**kw` is typed.
`**kw` is a dict by string-keyed dict definition and we are interested in the value type.
We would have to annotate every `**kw` as `dict[str, <thing we care about>]`.
It would be nice if numpydoc could use `https://pypi.org/project/sphinx-autodoc-typehints/`, then we could get rid of all type information for the Parameters / Return and just use the type hints.

Reply:
There is indeed an unfortunate discrepancy between how `**kwargs` are supposed to be typed and how they are supposed to be described in the docstring, but the docstring standard nonetheless only allows `dict, optional`.

Reply:
Changed. 😢